kinto/core/cache/memcached.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from functools import wraps
+from math import ceil, floor
+from time import time
+
+import memcache
+from pyramid.settings import aslist
+
+from kinto.core.cache import CacheBase
+from kinto.core.storage import exceptions
+from kinto.core.utils import json
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_memcached_error(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except TypeError:
+            raise
+        except Exception as e:
+            logger.exception(e)
+            raise exceptions.BackendError(original=e)
+    return wrapped
+
+
+def create_from_config(config, prefix=''):
+    """Redis client instantiation from settings.
+    """
+    settings = config.get_settings()
+    hosts = aslist(settings[prefix + 'hosts'])
+    return memcache.Client(hosts)
```

Comment:
This might be a silly question, but do we have to worry about reentrancy here (sharing the Memcache client across threads)? Or does uwsgi just spawn multiple threads with their own interpreters?

Reply:
The latter.